ruspy/simulation/simulation.py

```diff
@@ -51,51 +52,25 @@ def simulate(init_dict, ev_known=None, shock=None):
     beta = init_dict["beta"]
     num_periods = init_dict["periods"]
     params = np.array(init_dict["params"])
-    if "real_trans" in init_dict.keys():
-        real_trans = np.array(init_dict["real_trans"])
-    else:
-        real_trans = np.array(init_dict["known_trans"])
     if init_dict["maint_func"] == "linear":
         maint_func = lin_cost
     else:
         maint_func = lin_cost
-    unobs = get_unobs(shock, num_buses, num_periods)
-    increments = np.random.choice(
-        len(real_trans), size=(num_buses, num_periods), p=real_trans
-    )
-    if ev_known is not None:
-        # If there is already ev given, the auxiliary function is skipped and the
-        # simulation is executed with no further increases of the state space. This
-        # option is perfect if only one parameter in the setting is varied and
-        # therefore the highest achievable state can be guessed.
-        num_states = int(len(ev_known))
-        costs = cost_func(num_states, maint_func, params)
-        states = np.zeros((num_buses, num_periods), dtype=int)
-        decisions = np.zeros((num_buses, num_periods), dtype=int)
-        utilities = np.zeros((num_buses, num_periods), dtype=float)
-        states, decisions, utilities = simulate_strategy_loop_known(
-            num_buses,
-            states,
-            decisions,
-            utilities,
-            costs,
-            ev_known,
-            increments,
-            num_periods,
-            beta,
-            unobs,
+    if ev_known.shape[0] != real_trans_mat.shape[0]:
+        raise ValueError(
+            "The transition matrix and the expected value of the agent "
+            "need to have the same size."
         )
-    else:
-        known_trans = np.array(init_dict["known_trans"])
-        states, decisions, utilities, num_states = simulate_strategy(
-            known_trans,
-            increments,
-            num_buses,
-            num_periods,
-            params,
-            beta,
-            unobs,
-            maint_func,
+    num_states = ev_known.shape[0]
+    unobs = get_unobs(shock, num_buses, num_periods)
```

Comment:
I do not understand the `get_unobs` function, please upgrade it with better structure and comments.